lib/puppet/util/windows/adsi.rb

```diff
@@ -162,6 +162,7 @@ def commit
       begin
         native_user.SetInfo unless native_user.nil?
       rescue Exception => e
+        raise Puppet::Error.new( "Puppet is not able to create or delete Active Directory users with the user resource", e ) if /8007089A in Active Directory/ =~ e.message
```

Comment:
Do we want to catch the error, or catch the problem earlier (i.e. we know that `DOMAIN\User` is invalid input)